src/workflow.rs

```diff
@@ -71,18 +113,49 @@ impl Branches {
     }
 }
 
+impl std::fmt::Display for Branches {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        if !self.branches.is_empty() {
+            writeln!(f, "branches:")?;
+            write!(f, "{}", &serde_yaml::to_string(&self.branches).unwrap())?;
```

Comment:
I'm still relying on serde_yaml to do the heavy lifting but you could probably write your own stuff and get rid of the dependency (serde_yaml) entirely that is unmaintained anyway.

Reply:
I'll keep it that way for now and I explore the topic